hibernate-core/src/main/java/org/hibernate/query/sqm/produce/function/spi/StandardAnsiSqlSqmAggregationFunctionTemplates.java

```diff
@@ -81,12 +83,14 @@ protected SqmExpression generateSqmFunctionExpression(
 				AllowableFunctionReturnType impliedResultType) {
 			final SqmExpression argument = arguments.get( 0 );
 
+			// todo (6.0) : would be better to allow extension of this, probably through JavaTypeDescriptor
 			final Class argumentJavaType = argument.getExpressableType().getJavaType();
-			final boolean isFloatingPointNumber = Float.class.isInstance( argumentJavaType )
-					|| Double.class.isInstance( argumentJavaType );
-			final boolean needsCast = sqlCastTypeForFloatingPointArgTypes != null && !isFloatingPointNumber;
+			final boolean isFloatingPointNumber = Float.class.equals( argumentJavaType )
```

Comment:
Is it possible that we get the primitive class types `float` or `double` here?

Reply:
I do not think so